OpenRA.Mods.Common/Activities/Air/ReturnToBase.cs

```diff
@@ -108,7 +108,10 @@ public override Activity Tick(Actor self)
 
 				self.CancelActivity();
 				if (nearestAfld != null)
-					return ActivityUtils.SequenceActivities(new Fly(self, Target.FromActor(nearestAfld)), new FlyCircle(self));
+					return ActivityUtils.SequenceActivities(
+						new Fly(self, Target.FromActor(nearestAfld), WDist.Zero, plane.Info.WaitDistanceFromResupplyBase),
+						new FlyCircleTimed(plane.Info.NumberOfTicksToVerifyAvailableAirport, self),
+						new ReturnToBase(self));
 				else
 					return new FlyCircle(self);
```

Comment:
Please change this to `return NextActivity`. Otherwise (some of) the mission scripts break, and 2) if `NextActivity` is null, aircraft will end up circling by themselves anyway, thanks to `FlyAndContinueWithCirclesWhenIdle`.